chain/chain/src/flat_storage_resharder.rs

```diff
@@ -0,0 +1,949 @@
+        })?,
+        col::POSTPONED_RECEIPT => copy_kv_to_child(|raw_key: &[u8]| {
+            parse_account_id_from_trie_key_with_separator(
+                col::POSTPONED_RECEIPT,
+                raw_key,
+                ALL_COLUMNS_WITH_NAMES[col::POSTPONED_RECEIPT as usize].1,
+            )
+        })?,
+        col::DELAYED_RECEIPT_OR_INDICES => todo!(),
+        col::PROMISE_YIELD_INDICES => todo!(),
+        col::PROMISE_YIELD_TIMEOUT => todo!(),
+        col::PROMISE_YIELD_RECEIPT => todo!(),
+        col::BUFFERED_RECEIPT_INDICES => todo!(),
+        col::BUFFERED_RECEIPT => todo!(),
+        _ => unreachable!(),
+    }
+    Ok(())
+}
+
+/// Performs post-processing of shard splitting after all key-values have been moved from parent to children.
+/// `success` indicates whether or not the previous phase was successful.
+fn split_shard_task_postprocessing(resharder: FlatStorageResharderInner, success: bool) {
+    let (parent_shard, status) = get_parent_shard_and_status(&resharder);
+    let SplittingParentStatus { left_child_shard, right_child_shard, flat_head, .. } = status;
+    let flat_store = resharder.runtime.store().flat_store();
+    info!(target: "resharding", ?parent_shard, "flat storage shard split task: post-processing");
+
+    let mut store_update = flat_store.store_update();
+    if success {
+        // Split shard completed successfully.
+        // Parent flat storage can be deleted from the FlatStoreManager.
+        resharder
+            .runtime
+            .get_flat_storage_manager()
+            .remove_flat_storage_for_shard(parent_shard, &mut store_update)
+            .unwrap();
+        store_update.remove_flat_storage(parent_shard);
+        // Children must perform catchup.
+        for child_shard in [left_child_shard, right_child_shard] {
+            store_update.set_flat_storage_status(
+                child_shard,
+                FlatStorageStatus::Resharding(FlatStorageReshardingStatus::CatchingUp(
+                    flat_head.hash,
+                )),
+            );
+        }
+        // TODO(trisfald): trigger catchup
+    } else {
+        // We got an error or an interrupt request.
+        // Reset parent.
+        store_update.set_flat_storage_status(
+            parent_shard,
+            FlatStorageStatus::Ready(FlatStorageReadyStatus { flat_head }),
+        );
+        // Remove children shards leftovers.
+        for child_shard in [left_child_shard, right_child_shard] {
+            store_update.remove_flat_storage(child_shard);
+        }
+    }
+    store_update.commit().unwrap();
+    // Terminate the resharding event.
+    *resharder.resharding_event.lock().unwrap() = None;
+}
+
+/// Struct to describe, perform and track progress of a flat storage resharding.
+#[derive(Clone, Debug)]
+pub enum FlatStorageReshardingEvent {
+    /// Split a shard.
+    /// Includes the parent shard uid and the operation' status.
+    Split(ShardUId, SplittingParentStatus),
+}
```

Comment:
This is easy to confuse with the other event, how about renaming to `...Status` or similar?

Reply:
Renamed for more clarity